### PR TITLE
[v6r21] Docs: start howto section

### PR DESCRIPTION
**Pull request by chaen**
Replaces https://github.com/DIRACGrid/DIRAC/pull/3925/files

This is just the beginning of the HowTo we discussed long ago. @fstagni maybe that would be a good place to write (or at least link) the docs you wrote for `adding a site` or `adding a queue`
There is also a FileCatalog CLI tutorial which could be turned into howto. But I would rather have https://github.com/DIRACGrid/DIRAC/issues/1335 sorted out before.

@andresailer : would it be possible to generate an `ref anchor` in the generated script documentation such that the How-to could refer to them ? 

BEGINRELEASENOTES
*Docs
NEW: DataManagement How-To

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
@chaen yes one can add an anchor, but this will only happen for the integration branch

File: docs/source/index.rst [hunk 1]
```diff
@@ -49,10 +49,20 @@ or in this `presentation <https://indico.cern.ch/event/505613/contributions/2227
...
+===========
+
+A series of FAQ and HowTo is compiled for various use cases here :doc:`HowTo/index`
 
 Documentation sources
 =====================
 
+
+
+
```
Comment (fstagni) [inline docs/source/index.rst]: The table below has not been updated (the `toctree` abobe is hidden)

**Comment by atsareg** [comment]
I may be missed some discussion, but why not make the How To section part of the User Guide ?

**Comment by chaen** [comment]
I seem to remember we discussed that and agreed to do it in a separate section because the HowTo is not specific to users or admin.

File: docs/source/HowTo/DataManagement/metadata.rst [hunk 2]
```diff
@@ -63,22 +66,24 @@ used in data selections.
...
    /vo.formation.idgrilles.fr/user/a/atsareg/newDir/wms_output.py
 
-8.2 File Provenance Metadata
-----------------------------
 
-In the File Catalog you can declare ancestor files for a given file. This is often needed
-in order to keep track of the derived data provenance path. The ancestor declaration is done
-as following::
+How to declare file's ancestors
+------------------------------
```
Comment (andresailer) [inline docs/source/HowTo/DataManagement/metadata.rst]: There are not enough dashes

Comment (chaen) [reply docs/source/HowTo/DataManagement/metadata.rst]: Updated, thanks

Comment (chaen) [reply docs/source/HowTo/DataManagement/metadata.rst]: Shouldn't that create a warning btw ? I did not see any

Comment (andresailer) [reply docs/source/HowTo/DataManagement/metadata.rst]: No there was a warning:
```
source/HowTo/DataManagement/metadata.rst:84: WARNING: Title underline too short.
```
Just in this branch there are still too many...

**Comment by atsareg** [comment]
HowTo is related either to users, admins or developers. So, it is more appropriate to have a HowTo section in each of these dedicated guides. This will be easier for all the reader categories to look for relevant information

File: docs/source/index.rst (hunk 1 above)
Comment (fstagni) [reply docs/source/index.rst]: It would be needed yes, but let's wait to see how the discussion on this section goes.

**Comment by chaen** [comment]
I must have dreamt this discussion because I have a deja vu feeling :D 
Thing is that it all depends on what you call a user. is it a end user submitting analysis ? is it a data manager or production manager ? is it the admin that installs it ? Some commands are fit for all, and you do not want to repeat that. Though I could agree that dirac-admin-* commands should go to the administrator section. 
Also frankly, there are many things that I do not know where to put. For example: you want to explain that DIRAC uses X509 as authentication. Where do you put that ? in the admin docs ? In the user one ? In the developer docs ?

**Comment by atsareg** [comment]
First, administrators and developers are also end users in most cases. So, let's call users - end users for the sake of this discussion. Of course, there will be topics that you will hesitate where to put. Then the rule would be - "if hesitation, put into the Users' Guide How To". If it is clear where it belongs, then it should go to the relevant guide. In each guide's HowTo section we can also put a reference to other guides HowTo sections. So, a new user can easilyfind it out if not aware about other HowTo's from the very beginning.

**Comment by chaen** [comment]
I am doubtful that this will work on the long run and I fear that we will end up in the exact same mess we have now, but why not start like this, and eventually move it. What do other say ? 
My main argument for having it in a central place is always the same: CTRL + F will bring you where you need

**Comment by atsareg** [comment]
Introducing top level How To is more a mess I think. We could structure the documentation completely differently (e.g. Getting Started, Advanced, Reference and HowTo with User, Admin and Developer parts inside each large guide). This would also have its advantages and disadvantages. But we made the choice and I do not see why we should now change the general structure because of Ctrl+F vision of the things (then it would be ideal to put everything in just one large page).

**Comment by chaen** [comment]
No no, do not get me wrong, I do not want to change the overall division ! I actually think it is quite good. We even reasserted it when we had our dev discussion (https://indico.cern.ch/event/674501/)
It is just that I would not know where to look for certain things. For example, if I write "How to start a replication transformation", where should we put that ?

**Comment by atsareg** [comment]
This is certainly in the Users' Guide HowTo even without the general rule application (see above). Production managers are just power end users.

**Comment by andresailer** [comment]
just
```
cd docs
make singlehtml
```
:grinning: 

I think the individual guides need some restructuring to make it more obvious what can be found in the different subsections of those guides

**Comment by chaen** [comment]
:D 
Clearly, we need to do cleanup :)

**Comment by chaen** [comment]
OK I moved the HowTo in the user section

**Pull request merged**